yt_dlp/extractor/piramidetv.py

```diff
@@ -0,0 +1,113 @@
+        },
+    }, {
+        'url': 'https://piramide.tv/video/wcYn6li79NgN',
+        'info_dict': {
+            'id': 'wcYn6li79NgN',
+            'title': 'ACEPTO TENER UN BEBE CON MI NOVIA\u2026? | Parte 1',
+            'description': '',
+            'channel': 'ARTA GAME',
+            'channel_id': 'arta_game',
+        },
+        'playlist_count': 4,
+    }]
+
+    def _real_extract(self, url):
+        def extract_video(video_id, fatal=False):
+            video_data = self._download_json(f'https://hermes.piramide.tv/video/data/{video_id}',
+                                             video_id, fatal=fatal)
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                f'https://cdn.piramide.tv/video/{video_id}/manifest.m3u8', video_id, fatal=fatal)
+            video_info = {**traverse_obj(video_data, {
+                'id': ('video', 'id', {str}),
+                'title': ('video', 'title', {str}),
+                'description': ('video', 'description', {str}),
+                'thumbnail': ('video', 'media', 'thumbnail', {url_or_none}),
+                'channel': ('video', 'channel', 'name', {str}),
+                'channel_id': ('video', 'channel', 'id', {str}),
+                'timestamp': ('video', 'date', {unified_timestamp}),
+            }),
+                'formats': formats,
+                'subtitles': subtitles,
+                'webpage_url': f'https://piramide.tv/video/{video_id}',
+                'webpage_url_basename': video_id,
+            }
+            next_video_id = traverse_obj(video_data, ('video', 'next_video', 'id', {str}))
+            return video_info, next_video_id
+
+        video_id = self._match_id(url)
+        entries = []
+        while video_id is not None:
+            video, next_video = extract_video(video_id, (not entries))
+            if video.get('formats'):
+                entries.append(video)
+            video_id = next_video if next_video != video_id else None
+
+        if len(entries) == 1:
+            return entries[0]
+        elif len(entries) > 1:
+            return self.playlist_result(entries, **traverse_obj(entries[0], {
+                'id': ('id'),
+                'title': ('title'),
+                'description': ('description'),
+                'channel': ('channel'),
+                'channel_id': ('channel_id'),
+            }))
+        else:
+            return {'id': video_id}
+
+
+class PiramideTVChannelURLIE(InfoExtractor):
+    _VALID_URL = r'https?://piramide\.tv/channel/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://piramide.tv/channel/thekalo',
+        'playlist_count': 10,
+        'info_dict': {
+            'id': 'thekalo',
+            'title': 'thekalo',
+        },
+    }]
+
+    def _real_extract(self, url):
+        if query := self._match_id(url):
+            return self.url_result(url=f'piramidetvall:{query}', url_transparent=True)
+
+
+class PiramideTVChannelIE(SearchInfoExtractor):
```

Comment:
I don't entirely understand why you added a search extractor for the channel. I'll change this to be just a single extractor.